### DISCUSSION TITLE
NumberBox inheriting the default min and max of RangeBase

### DISCUSSION THREAD

**Discussion by RobertBeekman** [amwx/FluentAvalonia#373] [General]
Hello,

I did not want to create an issue for this, since it's perhaps intentional.
In my application I was facing issues with the NumberBox, which had started clamping values between 0 and 100. After some investigation I noticed that NumberBox inherits its min and max properties from RangeBase [here](https://github.com/amwx/FluentAvalonia/blob/6220d7d699d40108914b6685ee8a64de07e12d75/src/FluentAvalonia/UI/Controls/NumberBox/NumberBox.properties.cs#L80).

RangeBase in turn has a max of 100 as set [here](https://github.com/AvaloniaUI/Avalonia/blob/b2b908c6e89727e3f3d945288fa0706b245f778b/src/Avalonia.Controls/Primitives/RangeBase.cs#LL23C3-L23C3). I personally don't think these defaults are very appropriate for the numberbox. They make sense in a slider per example, but here not so much.

Perhaps it was unintentional, and perhaps you have a good reason for it I just don't know about, but figured I'd bring it up :)

PS, a simple workaround that I'm using right now:
```xaml
<Style Selector="controls|NumberBox">
    <Setter Property="Maximum" Value="{x:Static system:Double.MaxValue}"></Setter>
    <Setter Property="Minimum" Value="{x:Static system:Double.MinValue}"></Setter>
</Style>
```

**Comment by amwx** [comment]
Added an issue from this. Yep, was just an oversight when the properties switched from Direct->Styled. I checked WinUI and the history and it was previously correct at double.Min and double.Max
